src/system/SystemLayer.cpp

```diff
@@ -565,7 +565,7 @@ Error Layer::SetClock_RealTime(uint64_t newCurTime)
 void Layer::DispatchTimerCallbacks(const uint64_t kCurrentEpoch)
 {
     // dispatch TimerCallbacks
-    Cancelable ready = mTimerCallbacks.DequeueBy(TimerReady, (void *) &kCurrentEpoch);
+    Cancelable ready = mTimerCallbacks.DequeueBy(TimerReady, const_cast<void *>(reinterpret_cast<const void *>(&kCurrentEpoch)));
```

Comment:
Again, worth a comment explaining the `const_cast` (which is presumably that the callee just passes through the pointer opaquely and that `TimerReady` casts to a `const` thing).  

That said, `TimerReady` casts to `const Timer::Epoch *`, so could you please file a followup to make this function take `const Timer::Epoch` as an arg, not a raw `uint64_t`?

Reply:
Changed to Timer::Epoch and removed cast.